### PR TITLE
Update to last version of PHP, some parts fixed.

### PR DESCRIPTION
**Pull request by masterking32**

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
Thank you!!

Have you successfully made it work? we see people asking for this regularly in our discord (join us!!!)

**Comment by masterking32** [comment]
Not fully test, bcz I don't apply patch to server core. but I test it for some parts & a lot of errors about new PHP version & password hash method changed.

**Comment by BarbzYHOOL** [comment]
The SQL may have changed too

**Comment by masterking32** [comment]
> 
> 
> The SQL may have changed too

If have error share it. :)

**Comment by BarbzYHOOL** [comment]
@poszer come here ! :D

**Comment by deadmanevil** [comment]
when wanting to select the realm it does not leave it remains in small box and from there it does not pass more

**Comment by masterking32** [comment]
> 
> 
> when wanting to select the realm it does not leave it remains in small box and from there it does not pass more

Generate a SQL backup of your data in transfer tables & share.

**Comment by deadmanevil** [comment]
[authLkBackup 12-11-18.zip](https://github.com/azerothcore/web-character-migration-tool/files/2574131/authLkBackup.12-11-18.zip)
there is the backup of the sql

**Comment by deadmanevil** [comment]
[authLkBackup 13-11-18.zip](https://github.com/azerothcore/web-character-migration-tool/files/2574137/authLkBackup.13-11-18.zip)
this is the transfer table

**Comment by deadmanevil** [comment]
image

**Comment by masterking32** [comment]
So, I dont have data for test, just make a full backup with example data, for character & auth. To use your databases & data for fix them all.

**Comment by deadmanevil** [comment]
[charaLkBackup 13-11-18.zip](https://github.com/azerothcore/web-character-migration-tool/files/2578538/charaLkBackup.13-11-18.zip)
there's the backup

**Comment by deadmanevil** [comment]
<e>Query: INSERT INTO `characters` (`guid`,`account`,`name`,`race`,`class`,`gender`,`level`,`xp`,`money`,`playerBytes`,`playerBytes2`,`pla...

Error Code: 1054
Unknown column 'playerBytes' in 'field list'

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0.396 sec

**Comment by masterking32** [comment]
Hi, you share your auth with all account's information remove that! just need a row for example, but thank you I have this for test now remove it.
New pull added, test & share errors & send me data for this form to go next step.

sketch

**Comment by ELdoBA** [comment]
This should working
```
INSERT INTO `characters` (`guid`, `account`, `name`, `race`, `class`, `gender`, `level`, `xp`, `money`, `skin`, `face`, `hairStyle`, `hairColor`, `facialStyle`, `bankSlots`, `restState`, `playerFlags`, `position_x`, `position_y`, `position_z`, `map`, `instance_id`, `instance_mode_mask`, `orientation`, `taximask`, `online`, `cinematic`, `totaltime`, `leveltime`, `logout_time`, `is_logout_resting`, `rest_bonus`, `resettalents_cost`, `resettalents_time`, `trans_x`, `trans_y`, `trans_z`, `trans_o`, `transguid`, `extra_flags`, `stable_slots`, `at_login`, `zone`, `death_expire_time`, `taxi_path`, `arenaPoints`, `totalHonorPoints`, `todayHonorPoints`, `yesterdayHonorPoints`, `totalKills`, `todayKills`, `yesterdayKills`, `chosenTitle`, `knownCurrencies`, `watchedFaction`, `drunk`, `health`, `power1`, `power2`, `power3`, `power4`, `power5`, `power6`, `power7`, `latency`, `talentGroupsCount`, `activeTalentGroup`, `exploredZones`, `equipmentCache`, `ammoId`, `knownTitles`, `actionBars`, `grantableLevels`, `deleteInfos_Account`, `deleteInfos_Name`, `deleteDate`) VALUES (1000000, 1, 'CTrigger1', 7, 8, 1, 80, 0, 2137455906, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 647.644, 571, 0, 0, 0, '0 0 0 0 0 0 0 0 0 0 0 0 0 0', 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8, 4395, 0, NULL, 25, 31685, 0, 0, 36, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, NULL, NULL, 0, NULL, 0, 0, NULL, NULL, NULL);
INSERT INTO `characters` (`guid`, `account`, `name`, `race`, `class`, `gender`, `level`, `xp`, `money`, `skin`, `face`, `hairStyle`, `hairColor`, `facialStyle`, `bankSlots`, `restState`, `playerFlags`, `position_x`, `position_y`, `position_z`, `map`, `instance_id`, `instance_mode_mask`, `orientation`, `taximask`, `online`, `cinematic`, `totaltime`, `leveltime`, `logout_time`, `is_logout_resting`, `rest_bonus`, `resettalents_cost`, `resettalents_time`, `trans_x`, `trans_y`, `trans_z`, `trans_o`, `transguid`, `extra_flags`, `stable_slots`, `at_login`, `zone`, `death_expire_time`, `taxi_path`, `arenaPoints`, `totalHonorPoints`, `todayHonorPoints`, `yesterdayHonorPoints`, `totalKills`, `todayKills`, `yesterdayKills`, `chosenTitle`, `knownCurrencies`, `watchedFaction`, `drunk`, `health`, `power1`, `power2`, `power3`, `power4`, `power5`, `power6`, `power7`, `latency`, `talentGroupsCount`, `activeTalentGroup`, `exploredZones`, `equipmentCache`, `ammoId`, `knownTitles`, `actionBars`, `grantableLevels`, `deleteInfos_Account`, `deleteInfos_Name`, `deleteDate`) VALUES (2000000, 1, 'CTrigger2', 7, 8, 1, 80, 0, 2137455906, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 647.644, 571, 0, 0, 0, '0 0 0 0 0 0 0 0 0 0 0 0 0 0', 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8, 4395, 0, NULL, 25, 31685, 0, 0, 36, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, NULL, NULL, 0, NULL, 0, 0, NULL, NULL, NULL);

```

**Comment by masterking32** [comment]
Updated ...
I think everything is fine now, just test it & report bugs.

sketch

**Comment by ELdoBA** [comment]
Well done bro 👌👏

**Comment by BarbzYHOOL** [comment]
poszer has tested

@masterking32 awesome man

**Comment by ELdoBA** [comment]
@BarbzYhool The patch will need update, its old. Lets wait for it

**Comment by BarbzYHOOL** [comment]
Ok I suppose you talked in priv messages

**Comment by masterking32** [comment]
PHP script is fine now, but need update patch for core.

**Comment by deadmanevil** [comment]
image
to perform the installation not in local host does not let upload the file stays that way
or what configuration should have

**Comment by ELdoBA** [comment]
> hat configuration should have

Apply updates

**Comment by deadmanevil** [comment]
and the download ise configuration and so it remains with external ip and not in localhost

**Comment by ELdoBA** [comment]
@deadmanevil Do this :

```DELETE FROM `auth`.`account_transfer_blacklist` WHERE  `b_address`='127.0.0.1';```

```DELETE FROM `auth`.`account_transfer_blacklist` WHERE  `b_address`='localhost';```

**Comment by BarbzYHOOL** [comment]
So for the patch, what is needed?

**Comment by deadmanevil** [comment]
the above and I let it upload but now when I create an account do not log me in as if it were not

**Comment by deadmanevil** [comment]
and that sql must be put in character

**Comment by deadmanevil** [comment]
and I think also by means of merging an account and it does not let login and by means of the console by command and there if it lets logear    
also when I send the migration does not appear in administration to accept or not and the failure is that if you do with web fusion, you will not be allowed to log in and through the consulate if you do not know someone who can do a test with external ip that is not localhost bone in some vps to see the errors

**Comment by BarbzYHOOL** [comment]
no one understands what you mean, make an effort please

**Comment by ELdoBA** [comment]
> 
> 
> and I think also by means of merging an account and it does not let login and by means of the console by command and there if it lets logear
> also when I send the migration does not appear in administration to accept or not and the failure is that if you do with web fusion, you will not be allowed to log in and through the consulate if you do not know someone who can do a test with external ip that is not localhost bone in some vps to see the errors

Than you have done something wrong. I'm using it, and work fine

**Comment by deadmanevil** [comment]
> > y creo que también mediante la fusión de una cuenta y no permite el inicio de sesión y mediante la consola mediante un comando y si se permite iniciar sesión 
> > también cuando envío la migración no aparece en la administración para aceptarla o no y el error es que Si lo hace con web fusion, no se le permitirá iniciar sesión ni a través del consulado si no conoce a alguien que pueda hacer una prueba con una IP externa que no sea localhost bone en algunos vps para ver los errores.
> 
> Que has hecho algo malo. Lo estoy usando, y funciona bien.

you are using it in some vps or something like that to know if it is like that and also you can share the sql that you have to put in and updated to be abl

**Comment by BarbzYHOOL** [comment]
This needs an update apparently https://github.com/azerothcore/web-character-migration-tool/blob/master/SQL%2BCore's%20Commit/at_login.patch

**Comment by masterking32** [comment]
Finally, PHP files is ready, just need test new core patch.

**Comment by masterking32** [comment]
Not patch apply required. removed from readme file.

File: README.md [hunk 1]
```diff
@@ -5,20 +5,30 @@ v12 >> WOTLK (3.3.5a) >> WOTLK (3.3.5a)
...
+Import SQL files to your databases, Inside "SQL+Core's Commit".
+
+Config file "core/init.php" and "_transfer/t_config.php"
 
+In your database Table, 'account_transfer_blacklist' you can edit or add some servers to your transfer black list.
 
-More information and other items: ZoneWoW.com
+What is new?
+========
+Updated for support last version of PHP.
```
Comment (BarbzYHOOL) [inline README.md]:
```suggestion
Updated to support last version of PHP (5.3-7.x).
```

File: README.md [hunk 2]
```diff
@@ -5,20 +5,30 @@ v12 >> WOTLK (3.3.5a) >> WOTLK (3.3.5a)
...
 
-Desch
+Fix all bugs.
```
Comment (BarbzYHOOL) [inline README.md]:
```suggestion
Fixed all bugs up to AzerothCore rev d34e5329dfe6df44143219b3b9521610ee142ce2.
```

**Comment by BarbzYHOOL** [comment]
@deadmanevil test now :P

File: README.md (hunk 1 above)
Comment (masterking32) [reply README.md]: Thanks [25e038806fd3d66793922e36c473d8bf0132a561](https://github.com/masterking32/web-character-migration-tool/commit/25e038806fd3d66793922e36c473d8bf0132a561)

**Comment by deadmanevil** [comment]
> @deadmanevil test now :P

ok I would like to see if there are failures or not

**Comment by masterking32** [comment]
Done, Tested & completed.

**Comment by ELdoBA** [comment]
Tested, everything work perfectly. Thank you for this @masterking32. Amazing work!

**Review by STARRHELD — APPROVED** [review]
+

**Comment by BarbzYHOOL** [comment]
@masterking32 thank you very much

**Comment by deadmanevil** [comment]
> Done, Tested & completed.

image

do the test with a player of mine from another server and I got this

**Comment by deadmanevil** [comment]
> Done, Tested & completed.

because you think this happens to me this something is wrong with the tables someone already verified it from another server some migration

**Comment by BarbzYHOOL** [comment]
open another issue then and be precise because what you're saying is NOT helpful

**Pull request merged**